src/offset.js

```diff
@@ -82,27 +82,28 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
 			return;
 		}
 
-		docElem = doc.documentElement;
+		rect = elem.getBoundingClientRect();
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// Make sure element is not hidden (display: none) or disconnected
+		if ( !rect.width && !rect.height && !rect.left &&
+			!rect.top && !rect.right && !rect.bottom ) {
```

Comment:
In my personal opinion, I do not think what visually hidden element is always hidden by perspective of developer or even browser. If element participates in browser layout -- then that element absolutely not hidden. Consider following situation:

Someone uses empty ```<div>``` like a placeholder for future content (ajax loading for example). By some situation -- user scroll page so that position ```<div>``` becomes in view (or other same situation). And then developer needs to load content in that ```<div>```.
So, to determine this situation developer absolutely need to know position of that ```<div>```, even if height of that ```<div>``` is 0.

Try to run this example line by line in console ($0 === some block in this page):
```javascript
var a = document.createElement('div');
$0.appendChild(a);
a.getBoundingClientRect(); // ClientRect {height: 0, width: 653.6363525390625, left: 124.90056610107422, bottom: 405.553955078125, right: 778.5369262695312…}
a.style.margin = '20px';
a.getBoundingClientRect(); // ClientRect {height: 0, width: 613.6363525390625, left: 144.9005584716797, bottom: 425.553955078125, right: 758.5369262695312…}
```

As you can see, practically that element is not hidden (even if it's not visible by eye on a page). Even more, it still participates in layout and when you will add margin to element, it still will be hidden, but will take 40px height on a screen.

If this is "rare" case, then okay, I will simplify that condition. But for now I will wait for other responses.